sentinel/lib/messages.js

```diff
@@ -1,228 +1,115 @@
-        return utils.translate(configuration.translation_key, locale);
-    }
-
-    // otherwise, use the configured messages (deprecated)
-    var messages = configuration.messages || configuration.message;
-    if (!_.isArray(messages)) {
-        console.warn('Message property should be an array. Please check your configuration.');
-        return '';
-    }
-    if (!messages.length) {
-        console.warn('Message property array was empty. Please check your configuration.');
-        return '';
-    }
-    // default to first item in messages array in case locale match fails
-    var message = _.findWhere(messages, { locale: locale || 'en' }) || messages[0];
-    return (message.content && message.content.trim()) || '';
-}
+    messageUtils = require('./message-utils'),
+    config = require('../config');
 
 module.exports = {
-    /*
-     * Provide some extra template context, internal fields always override
-     * doc/form fields.
-     */
-    extractTemplateContext: function(doc) {
-        var clinic = utils.getClinic(doc);
-        var internal = {
-            contact: clinic && clinic.contact,
-            clinic: clinic,
-            parent: utils.getHealthCenter(doc),
-            health_center: utils.getHealthCenter(doc),
-            grandparent: utils.getDistrict(doc),
-            district: utils.getDistrict(doc)
+    addMessage: (doc, messageConfig, recipient = 'clinic', context = {}) => {
+        doc.tasks = doc.tasks || [];
+        const content = {
+            translationKey: messageConfig.translation_key,
+            message: messageConfig.message // deprecated usage
         };
-        return _.defaults(internal, doc.fields, doc);
-    },
-    scheduleMessage: function(doc, msg, phone, registrations, patient) {
-        var templateContext = extendedTemplateContext(doc, {
-            registrations: registrations,
-            patient: patient
-        });
-
-        phone = phone ? String(phone) : phone;
-
         try {
-            utils.addScheduledMessage(doc, {
-                due: msg.due,
-                message: template.render(msg.message, templateContext),
-                group: msg.group,
-                phone: phone,
-                type: msg.type,
-                translation_key: msg.translation_key
-            });
+            const messages = messageUtils.generate(
+                config.getAll(),
+                utils.translate,
+                doc,
+                content,
+                recipient,
+                context
+            );
+            const task = { messages: messages };
+            utils.setTaskState(task, module.exports.isOutgoingAllowed(doc.from) ? 'pending' : 'denied');
+            doc.tasks.push(task);
+            return task;
         } catch(e) {
             utils.addError(doc, {
-                message: e.message + ': ' + msg.message,
+                message: e.message + ': ' + messageConfig.translation_key || messageConfig.messages,
```

Comment:
An observation - in this bit of code we have three or more (different?) variables with very similar names:
* `task.messages`
* `e.message`
* `error.message`
* `messageConfig.messages`
* `messageUtils`
* `messages`
and this is then followed by a function called `getMessage()`.

If there were clearer names that these could be given, I think it could be helpful.

Reply:
Most of these are from the data structure or configuration so are really hard to change. The new module `messageUtils` could be changed easily, but I'm struggling to come up with a better name.